### PR TITLE
Include markers issue #5092

### PR DESCRIPTION
**Pull request by ImreC**
@matteius @oz123 I think this is literally what needs to happen for this, after the changes that were done a month ago. Or am I overlooking something?

### CODE REVIEW

**Comment by oz123** [comment]
@ImreC thanks for contributing again.
I would appreciate if it, if you could include a test case and the usual news fragment.

**Review by oz123 — CHANGES_REQUESTED** [review]
See my other comment about test and news.

**Comment by ImreC** [comment]
Yeah sure. Was just checking if this was literally it. Will do that later today

**Comment by matteius** [comment]
Yeah that seems like the logical change.  @ImreC Not sure you timeline, but I'd like to do another pipenv release due to another issue in the next week or so and would love if we could include this one as well.

**Comment by ImreC** [comment]
Hi @matteius, this completely fell off the map on my end. I'm having a very busy couple of weeks. I will find some time on Friday to do this

**Comment by ImreC** [comment]
@oz123 @matteius I have now implemented it as an optional flag. Do you agree with this approach? Happy to make it default if that's better

**Comment by matteius** [comment]
@ImreC I am thinking it should be the default behavior to include markers, but perhaps a flag like you have to exclude them (non-default) makes sense?  Especially with reports like this, having the ability to exclude markers may be relevant for some:

Also, Not sure if you saw my comment the other day I CC'd you on another issue report, the TLDR is (having trouble finding the ticket link)
> I do wonder if in the code I linked to for the requirements command, if we should have the logic first add the dev dependencies so that the production dependencies can overwrite any of those? -- that seems more correct to me than allowing the dev dependencies to break the production specifiers.

**Comment by oz123** [comment]
I tend to agree here, the flag should actually be `--exclude-markers` and the default behaviour should include them.

**Comment by ImreC** [comment]
@oz123 @matteius I flipped the markers flag. I really have been short for time the last couple of weeks. Hope this covers all.

File: docs/advanced.rst [hunk 1]
```diff
@@ -281,6 +281,7 @@ flag::
     pytest==3.2.3
 
 Adding the ``--hash`` flag will add package hashes to the output for extra security.
+Adding the ``--exclude-markers`` flagwill exclude the markers from the output.
```
Comment (matteius) [inline docs/advanced.rst:284]: typo need space `flag will`  -- I tried to edit this but I don't think I can directly without some hoop jumping.

File: pipenv/cli/command.py [hunk 2]
```diff
@@ -776,18 +777,18 @@ def requirements(state, dev=False, dev_only=False, hash=False):
 
     deps = {}
 
-    if not dev_only:
-        deps.update(lockfile["default"])
     if dev or dev_only:
         deps.update(lockfile["develop"])
+    if not dev_only:
```
Comment (matteius) [inline pipenv/cli/command.py:782]: Thanks for flipping the ordering here!

**Review by matteius — APPROVED** [review]
I approve @ImreC nice work -- just if you can please add the space for the typo so we can merge soon.

**Comment by oz123** [comment]
@ImreC @matteius please also fix the merge conflict.

**Comment by matteius** [comment]
@oz123 I am not seeing what you mean by a potential merge conflict ... If there is not one, I am ready to merge this and will push a commit to main to fix the documentation typo.

**Comment by oz123** [comment]
Ay! Sorry depending on the type of merge, there will be a conflict or there won't be.
image

image

**Pull request merged**